
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>snakemake/clindet: Citations &#8212; Clindet 0.1.0 documentation</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="_static/custom.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="_static/documentation_options.js?v=a1637f0b"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'CITATIONS';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="4. Use case III: Mutational signatures analysis for C.elegans" href="usecase/case_three.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/Clindet_logo.png" class="logo__image only-light" alt="Clindet 0.1.0 documentation - Home"/>
    <script>document.write(`<img src="_static/Clindet_logo.png" class="logo__image only-dark" alt="Clindet 0.1.0 documentation - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="index.html">
                    Clindet documentation
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="Quick/index.html">Quick Start</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Quick/intro.html">1. Clindet</a></li>
<li class="toctree-l2"><a class="reference internal" href="Quick/install.html">2. Setup Clindet</a></li>
<li class="toctree-l2"><a class="reference internal" href="Quick/use_case.html">3. Use case</a></li>
<li class="toctree-l2"><a class="reference internal" href="Quick/workflow.html">4. clindet workflow</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="DNAseq/index.html">DNAseq analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="DNAseq/WES/index.html">1. Whole exome sequence data analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WES/QC.html">1.1. WES</a></li>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WES/mutation.html">1.2. Call Mutation</a></li>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WES/CNV.html">1.3. Call CNV</a></li>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WES/SV.html">1.4. WES</a></li>
</ul>
</details></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="DNAseq/WGS/index.html">2. Whole genome sequence data analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WGS/QC.html">2.1. WGS</a></li>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WGS/mutation.html">2.2. Call Mutation</a></li>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WGS/CNV.html">2.3. Call CNV</a></li>
<li class="toctree-l3"><a class="reference internal" href="DNAseq/WGS/SV.html">2.4. WGS SV (structural variants calling)</a></li>
</ul>
</details></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="RNAseq/index.html">RNAseq analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="RNAseq/qc.html">1. QC</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq/expression.html">2. Expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="RNAseq/fusion.html">3. Gene  fusion</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="usecase/index.html">Use Case</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="usecase/intro.html">1. Real world example</a></li>
<li class="toctree-l2"><a class="reference internal" href="usecase/case_one.html">2. Use case I: SNV and CNV calling from Whole exome sequencing data</a></li>
<li class="toctree-l2"><a class="reference internal" href="usecase/case_two.html">3. Use case II: Fusion genes detection from multiple myeloma patient RNA-seq</a></li>
<li class="toctree-l2"><a class="reference internal" href="usecase/case_three.html">4. Use case III: Mutational signatures analysis for <em><strong>C.elegans</strong></em></a></li>
</ul>
</details></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">snakemake/clindet: Citations</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/CITATIONS.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>snakemake/clindet: Citations</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#snakemake">snakemake</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pipeline-tools">Pipeline tools</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#software-packaging-containerisation-tools">Software packaging/containerisation tools</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="snakemake-clindet-citations">
<h1>snakemake/clindet: Citations<a class="headerlink" href="#snakemake-clindet-citations" title="Link to this heading">#</a></h1>
<section id="snakemake">
<h2><a class="reference external" href="https://f1000research.com/articles/10-33/v2">snakemake</a><a class="headerlink" href="#snakemake" title="Link to this heading">#</a></h2>
<blockquote>
<div><p>Ewels PA, Peltzer A, Fillinger S, Patel H, Alneberg J, Wilm A, Garcia MU, Di Tommaso P, Nahnsen S. The nf-core framework for community-curated bioinformatics pipelines. Nat Biotechnol. 2020 Mar;38(3):276-278. doi: 10.1038/s41587-020-0439-x. PubMed PMID: 32055031.</p>
</div></blockquote>
</section>
<section id="pipeline-tools">
<h2>Pipeline tools<a class="headerlink" href="#pipeline-tools" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/gigascience/giab008">BCFtools</a></p></li>
</ul>
<blockquote>
<div><p>Danecek, P., Bonfield, J. K., Liddle, J., Marshall, J., Ohan, V., Pollard, M. O., Whitwham, A., Keane, T., McCarthy, S. A., Davies, R. M., &amp; Li, H. (2021). Twelve years of SAMtools and BCFtools. GigaScience, 10(2), giab008. <a class="reference external" href="https://doi.org/10.1093/gigascience/giab008">https://doi.org/10.1093/gigascience/giab008</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btp324">BWA</a></p></li>
</ul>
<blockquote>
<div><p>Li, H., &amp; Durbin, R. (2009). Fast and accurate short read alignment with Burrows–Wheeler transform. Bioinformatics, 25(14), 1754–1760. <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btp324">https://doi.org/10.1093/bioinformatics/btp324</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1109/IPDPS.2019.00041">bwa-mem2</a></p></li>
</ul>
<blockquote>
<div><p>Vasimuddin, Md., Misra, S., Li, H., &amp; Aluru, S. (2019). Efficient Architecture-Aware Acceleration of BWA-MEM for Multicore Systems. 2019 IEEE International Parallel and Distributed Processing Symposium (IPDPS), 314–324. <a class="reference external" href="https://doi.org/10.1109/IPDPS.2019.00041">https://doi.org/10.1109/IPDPS.2019.00041</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1038/s41467-020-19406-4">CHORD</a></p></li>
</ul>
<blockquote>
<div><p>Nguyen, L., W. M. Martens, J., Van Hoeck, A., &amp; Cuppen, E. (2020). Pan-cancer landscape of homologous recombination deficiency. Nature Communications, 11(1), 5584. <a class="reference external" href="https://doi.org/10.1038/s41467-020-19406-4">https://doi.org/10.1038/s41467-020-19406-4</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/bty560">fastp</a></p></li>
</ul>
<blockquote>
<div><p>Chen, S., Zhou, Y., Chen, Y., &amp; Gu, J. (2018). fastp: An ultra-fast all-in-one FASTQ preprocessor. Bioinformatics, 34(17), i884–i890. <a class="reference external" href="https://doi.org/10.1093/bioinformatics/bty560">https://doi.org/10.1093/bioinformatics/bty560</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btp324">GATK</a></p></li>
</ul>
<blockquote>
<div><p>McKenna, A., Hanna, M., Banks, E., Sivachenko, A., Cibulskis, K., Kernytsky, A., Garimella, K., Altshuler, D., Gabriel, S., Daly, M., &amp; DePristo, M. A. (2010). The Genome Analysis Toolkit: A MapReduce framework for analyzing next-generation DNA sequencing data. Genome Research, 20(9), 1297–1303. <a class="reference external" href="https://doi.org/10.1101/gr.107524.110">https://doi.org/10.1101/gr.107524.110</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1186/s13059-021-02423-x">GRIDSS2</a></p></li>
</ul>
<blockquote>
<div><p>Cameron, D. L., Baber, J., Shale, C., Valle-Inclan, J. E., Besselink, N., van Hoeck, A., Janssen, R., Cuppen, E., Priestley, P., &amp; Papenfuss, A. T. (2021). GRIDSS2: Comprehensive characterisation of somatic structural variation using single breakend variants and structural variant phasing. Genome Biology, 22(1), Article 1. <a class="reference external" href="https://doi.org/10.1186/s13059-021-02423-x">https://doi.org/10.1186/s13059-021-02423-x</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1038/s41588-023-01367-1">LILAC</a></p></li>
</ul>
<blockquote>
<div><p>Martínez-Jiménez, F., Priestley, P., Shale, C., Baber, J., Rozemuller, E., &amp; Cuppen, E. (2023). Genetic immune escape landscape in primary and metastatic cancer. Nature Genetics, 55(5), 820–831. <a class="reference external" href="https://doi.org/10.1038/s41588-023-01367-1">https://doi.org/10.1038/s41588-023-01367-1</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1016/j.xgen.2022.100112">LINX</a></p></li>
</ul>
<blockquote>
<div><p>Shale, C., Cameron, D. L., Baber, J., Wong, M., Cowley, M. J., Papenfuss, A. T., Cuppen, E., &amp; Priestley, P. (2022). Unscrambling cancer genomes via integrated analysis of structural variation and copy number. Cell Genomics, 2(4). <a class="reference external" href="https://doi.org/10.1016/j.xgen.2022.100112">https://doi.org/10.1016/j.xgen.2022.100112</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1038/s41586-019-1689-y">PURPLE</a></p></li>
</ul>
<blockquote>
<div><p>Priestley, P., Baber, J., Lolkema, M. P., Steeghs, N., de Bruijn, E., Shale, C., Duyvesteyn, K., Haidari, S., van Hoeck, A., Onstenk, W., Roepman, P., Voda, M., Bloemendal, H. J., Tjan-Heijnen, V. C. G., van Herpen, C. M. L., Labots, M., Witteveen, P. O., Smit, E. F., Sleijfer, S., … Cuppen, E. (2019). Pan-cancer whole-genome analyses of metastatic solid tumours. Nature, 575(7781), 210–216. <a class="reference external" href="https://doi.org/10.1038/s41586-019-1689-y">https://doi.org/10.1038/s41586-019-1689-y</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btv098">Sambamba</a></p></li>
</ul>
<blockquote>
<div><p>Tarasov, A., Vilella, A. J., Cuppen, E., Nijman, I. J., &amp; Prins, P. (2015). Sambamba: Fast processing of NGS alignment formats. Bioinformatics, 31(12), 2032–2034. <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btv098">https://doi.org/10.1093/bioinformatics/btv098</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/gigascience/giab008">SAMtools</a></p></li>
</ul>
<blockquote>
<div><p>Danecek, P., Bonfield, J. K., Liddle, J., Marshall, J., Ohan, V., Pollard, M. O., Whitwham, A., Keane, T., McCarthy, S. A., Davies, R. M., &amp; Li, H. (2021). Twelve years of SAMtools and BCFtools. GigaScience, 10(2), giab008. <a class="reference external" href="https://doi.org/10.1093/gigascience/giab008">https://doi.org/10.1093/gigascience/giab008</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/bts635">STAR</a></p></li>
</ul>
<blockquote>
<div><p>Dobin, A., Davis, C. A., Schlesinger, F., Drenkow, J., Zaleski, C., Jha, S., Batut, P., Chaisson, M., &amp; Gingeras, T. R. (2013). STAR: Ultrafast universal RNA-seq aligner. Bioinformatics, 29(1), 15–21. <a class="reference external" href="https://doi.org/10.1093/bioinformatics/bts635">https://doi.org/10.1093/bioinformatics/bts635</a></p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btab343">VIRUSBreakend</a></p></li>
</ul>
<blockquote>
<div><p>Cameron, D. L., Jacobs, N., Roepman, P., Priestley, P., Cuppen, E., &amp; Papenfuss, A. T. (2021). VIRUSBreakend: Viral Integration Recognition Using Single Breakends. Bioinformatics, 37(19), 3115–3119. <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btab343">https://doi.org/10.1093/bioinformatics/btab343</a></p>
</div></blockquote>
</section>
<section id="software-packaging-containerisation-tools">
<h2>Software packaging/containerisation tools<a class="headerlink" href="#software-packaging-containerisation-tools" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://anaconda.com">Anaconda</a></p></li>
</ul>
<blockquote>
<div><p>Anaconda Software Distribution. Computer software. Vers. 2-2.4.0. Anaconda, Nov. 2016. Web.</p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/29967506/">Bioconda</a></p></li>
</ul>
<blockquote>
<div><p>Grüning B, Dale R, Sjödin A, Chapman BA, Rowe J, Tomkins-Tinch CH, Valieris R, Köster J; Bioconda Team. Bioconda: sustainable and comprehensive software distribution for the life sciences. Nat Methods. 2018 Jul;15(7):475-476. doi: 10.1038/s41592-018-0046-7. PubMed PMID: 29967506.</p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/28379341/">BioContainers</a></p></li>
</ul>
<blockquote>
<div><p>da Veiga Leprevost F, Grüning B, Aflitos SA, Röst HL, Uszkoreit J, Barsnes H, Vaudel M, Moreno P, Gatto L, Weber J, Bai M, Jimenez RC, Sachsenberg T, Pfeuffer J, Alvarez RV, Griss J, Nesvizhskii AI, Perez-Riverol Y. BioContainers: an open-source and community-driven framework for software standardization. Bioinformatics. 2017 Aug 15;33(16):2580-2582. doi: 10.1093/bioinformatics/btx192. PubMed PMID: 28379341; PubMed Central PMCID: PMC5870671.</p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://dl.acm.org/doi/10.5555/2600239.2600241">Docker</a></p></li>
</ul>
<blockquote>
<div><p>Merkel, D. (2014). Docker: lightweight linux containers for consistent development and deployment. Linux Journal, 2014(239), 2. doi: 10.5555/2600239.2600241.</p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/28494014/">Singularity</a></p></li>
</ul>
<blockquote>
<div><p>Kurtzer GM, Sochat V, Bauer MW. Singularity: Scientific containers for mobility of compute. PLoS One. 2017 May 11;12(5):e0177459. doi: 10.1371/journal.pone.0177459. eCollection 2017. PubMed PMID: 28494014; PubMed Central PMCID: PMC5426675.</p>
</div></blockquote>
</section>
</section>


                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="usecase/case_three.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title"><span class="section-number">4. </span>Use case III: Mutational signatures analysis for <em><strong>C.elegans</strong></em></p>
      </div>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#snakemake">snakemake</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pipeline-tools">Pipeline tools</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#software-packaging-containerisation-tools">Software packaging/containerisation tools</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Yuliang Zhang
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2024, Yuliang Zhang.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>